### PR TITLE
bpo-33361: Fix bug with seeking in StreamRecoders

### PR DESCRIPTION
**Pull request by ammaraskar**
The test is a basic example from the bpo ticket, more can be added if necessary.

https://bugs.python.org/issue33361

### CODE REVIEW

File: Lib/codecs.py [hunk 1]
```diff
@@ -847,6 +847,12 @@ def reset(self):
         self.reader.reset()
         self.writer.reset()
 
+    def seek(self, offset, whence=0):
+        # Seeks must be propogated to both the readers and writers
```
Comment (libcthorne) [inline Lib/codecs.py]: Minor nitpick but there is a typo in the comment ("propogated" -> "propagated")

Comment (ammaraskar) [reply Lib/codecs.py]: Thanks, fixed.

**Review by BoboTiG — APPROVED** [review]
Thank you for the patch :)

**Review by berkerpeksag — APPROVED** [review]

**Comment by berkerpeksag** [comment]
Thanks!

**Pull request merged**